containers/src/Kokkos_DualView.hpp

```diff
@@ -208,6 +208,12 @@ class DualView : public ViewTraits<DataType, Properties...> {
   using t_modified_flags = View<unsigned int[2], LayoutLeft, Kokkos::HostSpace>;
   t_modified_flags modified_flags;
 
+ public:
+  // does the DualView have only one device
+  static constexpr bool impl_dualview_stores_single_view =
+      SpaceAccessibility<Kokkos::HostSpace,
+                         typename t_dev::memory_space>::accessible;
+
```

Comment:
what happens if I use a host view and a shared space view?

Reply:
How many allocations `DualView` uses is given by it's memory space. Both for `HostSpace` and `SharedSpace`, only one allocation is referenced. If you the construct a `DualView` from two `View`s, one in `HostSpace` and the other one in `SharedSpace`, it only makes sense if they happen to point to the same memory (since `sync` and `modify` become no-ops). Whether that's possible or not is not a question for this pull request but rather for #7662.